components/advanced/babel-plugin.js

```diff
@@ -0,0 +1,165 @@
+import React from 'react';
+import SectionLayout from '../SectionLayout';
+import CodeBlock from '../CodeBlock';
+import Code from '../Code';
+import Note from '../Note';
+
+const installNPM = `
+npm install --save-dev babel-plugin-styled-components
+`.trim();
+
+const usage = `
+{
+  "plugins": ["styled-components"]
+}
+`.trim();
+
+const ssr = `
+{
+  "plugins": [
+    ["styled-components", {
+      "ssr": true
+    }]
+  ]
+}
+`.trim();
+
+const displayName = `
+{
+  "plugins": [
+    ["styled-components", {
+      "displayName": false
+    }]
+  ]
+}
+`.trim();
+
+const preprocess = `
+{
+  "plugins": [
+    ["styled-components", {
+      "preprocess": true
+    }]
+  ]
+}
+`.trim();
+
+const minify = `
+{
+  "plugins": [
+    ["styled-components", {
+      "minify": false
+    }]
+  ]
+}
+`.trim();
+
+const BabelPlugin = () =>
+  <SectionLayout title="Babel Plugin" labels={['v2']}>
+    <p>
+      This plugin adds support for server-side rendering, for minification of
+      styles and gives you a nicer debugging experience when using{' '}
+      <Code>styled-components</Code>.
+    </p>
+    <p>Usage</p>
+    <CodeBlock code={installNPM} language="node" />
+    <p>
+      Then in your babel configuration (probably <Code>.babelrc</Code>):
+    </p>
+    <CodeBlock code={usage} language="node" />
+    <SectionLayout sub title="Server-side rendering">
+      <Note>This option is turned off by default</Note>
+      <p>
+        By adding a unique identifier to every styled component this plugin
+        avoids checksum mismatches due to different class generation on the
+        client and on the server. If you do not use this plugin and try to
+        server-side render <Code>styled-components</Code> React will complain.
+      </p>
+      <p>
+        If you want server-side rendering support you can enable it with the{' '}
+        <Code>ssr</Code> option:
+      </p>
+      <CodeBlock code={ssr} language="node" />
+    </SectionLayout>
+    <SectionLayout sub title="Better debugging">
+      <p>
+        This babel plugin adds the components' name to the class name attached
+        to the DOM node. In your browsers DevTools you'll see:{' '}
+        <Code>
+          &lt;button class=&quot;sc-Button-asdf123 asdf123&quot; /&gt;
+        </Code>{' '}
+        instead of just <Code>&lt;button class=&quot;asdf123&quot; /&gt;</Code>.
+      </p>
+      <p>
+        This also adds support for showing your components' real name in the
+        React DevTools. They will normally show{' '}
+        <Code>&lt;styled.button&gt;</Code> for all of your components, but with
```

Comment:
The examples here are nice, but they're missing reference code.

It should probably intro with:

> Consider writing a styled component that renders a `button` element, called `MyButton`. It will normally show up as [...], but with [...]